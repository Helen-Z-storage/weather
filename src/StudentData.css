
.data_space {
    width: 100%; /* Full-width */
}

img {
  width: 7em;
  margin: 10px;
  border-radius: 50%;
  border: 1px solid #DCDCDC;
}

.stu_name {
    font-size: 2em;
    font-weight: bolder;
    float: left;
    padding: 0.3em 0em;
  }
  
.stu_data {
    padding-top: 5px;
  }
  