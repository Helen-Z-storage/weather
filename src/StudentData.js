import React from 'react';
import './StudentData.css';

function StudentData(props) {
    const {student, name, avg} = props;

    return (
        <tr>
            <td>
                <img alt={`${name} profile`} src={student.pic} />
            </td>
            <td>
                <ul className='data_space'>
                    <li key="0"><div className='stu_name' style={{width:"100%"}}>{name}</div></li>
                    <li key="1" className='stu_data'>{`Email: ${student.email}`}</li>
                    <li key="2" className='stu_data'>{`Company: ${student.company}`}</li>
                    <li key="3" className='stu_data'>{`Skill: ${student.skill}`}</li>
                    <li key="4" className='stu_data'>{`Average: ${Math.round(avg * 1000) / 1000}%`}</li>
                </ul>
            </td>
        </tr>
    );
}

export default StudentData;
