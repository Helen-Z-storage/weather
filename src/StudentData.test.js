import React from 'react'
import { create } from "react-test-renderer";

import '@testing-library/jest-dom/extend-expect' // expect function

import StudentData from './StudentData';

// required prop data of create component
const student = {
    city: "FushÃ«-Muhurr",
    company: "Yadel",
    email: "iorton0@imdb.com",
    firstName: "Ingaberg",
    grades: ["78", "100", "92", "86", "89", "88", "91", "87"],
    id: "1",
    lastName: "Orton",
    pic: "https://storage.googleapis.com/hatchways-app.appspot.com/" + 
            "ssessments/data/frontend/images/voluptasdictablanditiis.jpg",
    skill: "Oracle",
    tags: " "
};
const name = `${student.firstName} ${student.lastName}`.toUpperCase();
const avg = student.grades.map(grade => parseInt(grade, 10))
                            .reduce((a, b) => a + b, 0)
                            / student.grades.length;

describe('testing StudentData component', () => {
    // snampshot testing, for non-state component only
    test("Matches the snapshot", () => {
        // create component
        const studentData = create(
            <StudentData student={student} name={name} avg={avg}/>
        );

        // expecting output
        expect(studentData.toJSON()).toMatchSnapshot();
    });
});