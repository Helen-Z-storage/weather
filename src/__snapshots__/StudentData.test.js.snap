// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing StudentData component Matches the snapshot 1`] = `
<tr>
  <td>
    <img
      alt="INGABERG ORTON profile"
      src="https://storage.googleapis.com/hatchways-app.appspot.com/ssessments/data/frontend/images/voluptasdictablanditiis.jpg"
    />
  </td>
  <td>
    <ul
      className="data_space"
    >
      <li>
        <div
          className="stu_name"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          INGABERG ORTON
        </div>
      </li>
      <li
        className="stu_data"
      >
        Email: iorton0@imdb.com
      </li>
      <li
        className="stu_data"
      >
        Company: Yadel
      </li>
      <li
        className="stu_data"
      >
        Skill: Oracle
      </li>
      <li
        className="stu_data"
      >
        Average: 88.875%
      </li>
    </ul>
  </td>
</tr>
`;
